operator-framework-quarkus-extension/deployment/src/main/java/io/javaoperatorsdk/quarkus/extension/deployment/QuarkusExtensionProcessor.java

```diff
@@ -0,0 +1,172 @@
+            .done());
+
+    additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(OperatorProducer.class));
+  }
+
+  private ControllerConfiguration createControllerConfiguration(
+      ClassInfo info,
+      ClassOutput classOutput,
+      BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+    // first retrieve the custom resource class
+    final var rcInterface =
+        info.interfaceTypes().stream()
+            .filter(t -> t.name().equals(RESOURCE_CONTROLLER))
+            .findFirst()
+            .map(Type::asParameterizedType)
+            .orElseThrow(); // shouldn't happen since we're only dealing with ResourceController
+    // implementors already
+    final var crType = rcInterface.arguments().get(0).name().toString();
+
+    // create ResourceController bean
+    final var resourceControllerClassName = info.name().toString();
+    additionalBeans.produce(AdditionalBeanBuildItem.unremovableOf(resourceControllerClassName));
+
+    // generate associated Doneable class
+    final var doneableClassName = crType + "Doneable";
+    final var crDoneableClassName = CustomResourceDoneable.class.getName();
+    try (ClassCreator cc =
+        ClassCreator.builder()
+            .classOutput(classOutput)
+            .className(doneableClassName)
+            .superClass(crDoneableClassName)
+            .build()) {
+
+      MethodCreator ctor = cc.getMethodCreator("<init>", void.class.getName(), crType);
+      ctor.setModifiers(Modifier.PUBLIC);
+      final var functionName = Function.class.getName();
+      ctor.invokeSpecialMethod(
+          MethodDescriptor.ofConstructor(crDoneableClassName, crType, functionName),
+          ctor.getThis(),
+          ctor.getMethodParam(0),
+          ctor.invokeStaticMethod(
+              MethodDescriptor.ofMethod(functionName, "identity", functionName)));
+    }
+
+    // generate configuration
+    final var controllerAnnotation = info.classAnnotation(CONTROLLER);
+    final var crdName =
+        valueOrDefault(
+            controllerAnnotation, "crdName", AnnotationValue::asString, EXCEPTION_SUPPLIER);
+    final var configuration =
+        new QuarkusControllerConfiguration(
+            valueOrDefault(
+                controllerAnnotation,
+                "name",
+                AnnotationValue::asString,
+                () -> ControllerUtils.getDefaultResourceControllerName(info.simpleName())),
+            crdName,
+            valueOrDefault(
+                controllerAnnotation,
+                "finalizerName",
+                AnnotationValue::asString,
+                () -> ControllerUtils.getDefaultFinalizerName(crdName)),
+            valueOrDefault(
+                controllerAnnotation,
+                "generationAwareEventProcessing",
+                AnnotationValue::asBoolean,
+                () -> true),
+            valueOrDefault(
+                controllerAnnotation, "isClusterScoped", AnnotationValue::asBoolean, () -> false),
```

Comment:
is `isClusterScoped` used anywhere?

Reply:
It's used when the controller is registered with the Operator to know under which scope to register (namespace vs. cluster).

Reply:
Or at least, it should be but, indeed, it doesn't appear to be used at the moment. I'd keep it, though, because that's an option that we'd want to enable for controllers. That said, I'd rather fix this issue (and possibly others) into subsequent PRs than keeping this one open any longer. It's a pain to maintain and a huge Damocles' sword over people's head :)

Reply:
Can you point me to the code? Couldn't find it in the changeset.

Reply:
> Or at least, it should be but, indeed, it doesn't appear to be used at the moment. I'd keep it, though, because that's an option that we'd want to enable for controllers. That said, I'd rather fix this issue (and possibly others) into subsequent PRs than keeping this one open any longer. It's a pain to maintain and a huge Damocles' sword over people's head :)

As an alternative, it can be removed from the code and be addressed as part of the relevant issue. 
IMHO it Would be good practice to develop a feature that has **definition of done**.